app/rankers/algorithms.py

```diff
@@ -63,8 +77,8 @@ def blocklist(recs: List['RecommendationModel'], blocklist: List[str] = None) ->
 
 
 def thompson_sampling(
-        recs: List['RecommendationModel'],
-        clk_data: Dict[(int or str), 'ClickdataModel']) -> List['RecommendationModel']:
+        recs: RankableListType,
+        metrics: Dict[(int or str), 'MetricsModel']) -> RankableListType:
```

Comment:
This type annotation isn't valid.
```
In [1]: def f(a:(int or str)):
   ...:     return a
In [4]: typing.get_type_hints(f)
Out[4]: {'a': int}
```
```suggestion
        metrics: Dict[Union[int, str], 'MetricsModel']) -> RankableListType:
```